test/all.sh

```diff
@@ -47,9 +62,15 @@ fi
 function do_script() {
     script=$1
     $SCRIPT_LOC/${script}.sh
+    X=$?
+    update_exit_code_if_non_zero $X
     if [[ "$TEST_TARGET" = "x86-64-linux" && $DEFAULT_MODES = 1 ]]; then
         TEST_MODE=jit $SCRIPT_LOC/${script}.sh
+        X=$?
+        update_exit_code_if_non_zero $X
```

Comment:
I think there is a way to do this with `&&` in bash.

Reply:
Note sure what you mean. The point of the `X=$?` stuff is to make the CI fail if any of the tests fail. `&&` is generally used for chaining?

Reply:
I just mean, shortening up the code a little by not needing a whole `if` cascade. I think this might work:

`<shell command> || SCRIPT_EXIT_CODE=$?`

Reply:
Ah right. I wanted to preserve the first error code as opposed to the last one, which is why I did this. Up-to-you if you want to preserve the first error code or the last one.

Reply:
Why not just exit early on the first error?

Reply:
Hm, welll you get more data to work with if something fails. But yeah, we could also just do exit at first failure.

Reply:
I think it's better to fail at first error in CI; that way you can always scroll to the end of the log to see what failed. (Come to think of it, it might be worth doing the same for the Virgil CI).

Reply:
Right that's fair. I'll change it to fail on first error

Reply:
Done.